receiver/mongodbreceiver/metadata.yaml

```diff
@@ -1 +1,143 @@
 name: mongodbreceiver
+
+attributes:
+  database:
+    description: The name of a database.
+  memory_type:
+    description: The type of memory used.
+    enum:
+      - resident
+      - virtual
+      - mapped
+      - mapped_with_journal
```

Comment:
Unfortunately, I don't think we can apply these to a sum metric, since the sum is not meaningful. 

There's been some debate about whether `resident` and `virtual` are meaningfully summed. A quick audit of the repo shows that they typically are, and this is correct in my opinion. Also, there's nothing in the specification that indicates they should be separated. So, I think you can have `mongodb.memory.usage` with `type = resident | virtual`.

As I understand it, `mapped` and `mapped_with_journal` are overlapping, with each other, and with resident/virtual. To capture these then, we definitely need at least one additional metric. If you want to leave the values as is, then we'll need one metric for each. Another option, if it makes sense in the domain, is to derive a value from the two that removes overlap. Possibly, `mongodb.memory.mapped` with `type = mapped | journal`, where `journal = mapped_with_journal - mapped`. Depends if this formula is accurate and if such a `journal` value is meaningful.

Reply:
Hmm yes I agree they mapped and mapped_with_journal do not easily fit and don't mean much aggregated. 

My next question would be whether or not mapped and journaled would be their own metrics. The concept of journaling is in the event of an unexpected shutdown of mongod process, then it can be used to recover but this process requires memory. One helpful link I found on describing the places of storage engines and the journaling process across MMap and WiredTiger can be found [here](https://www.kenwalger.com/blog/nosql/mongodb/mongodb-storage-engine-journaling/). 

In believe that for right now, it makes the most sense scrap out the `mapped` and `mapped_with_journal` concept in favor of a more easily digestable `mongodb.memory.usage` metric and if the need comes to know the granularity of how much memory journaling or memory mapping is then we can add them later as more dedicated metrics. Do you have any thoughts on that approach?

Reply:
👍  Seems reasonable to me